kahypar/io/hypergraph_io.h

```diff
@@ -37,38 +38,106 @@ namespace kahypar {
 namespace io {
 using Mapping = std::unordered_map<HypernodeID, HypernodeID>;
 
-static inline void readHGRHeader(std::ifstream& file, HyperedgeID& num_hyperedges,
-                                 HypernodeID& num_hypernodes, HypergraphType& hypergraph_type) {
-  std::string line;
+class CheckedIStream {
+ public:
+  explicit CheckedIStream(const std::string& line, size_t line_number = 0):
+    _current(line.c_str()), _next(nullptr), _end(line.c_str() + line.size()), _line_number(line_number) {
+      // whitespace at end of line is not considered an error
+      while (_end > _current && *(_end - 1) == ' ') {
+        --_end;
+      }
+    }
+
+  template<typename ResultT>
+  bool operator>>(ResultT& result) {
+    unsigned long long val = std::strtoull(_current, &_next, 10);
+    if (val == 0 && _next == _current && _next != _end) {
+      std::cerr << "Error: Expected positive number";
```

Comment:
Can we use the `ERROR` instead of `std::err` macros inside `CheckedIStream` as well?

Reply:
Good point, done